luigi/worker.py

```diff
@@ -842,7 +842,14 @@ def _add(self, task, is_complete):
                     task.trigger_event(Event.DEPENDENCY_DISCOVERED, task, d)
                     yield d  # return additional tasks to add
 
-                deps = [d.task_id for d in deps]
+                # if dependencies check took too long, the parent task might have been completed by other worker
+                if status == PENDING and task.complete():
```

Comment:
Why is `status == PENDING` needed?  The comment doesn't cover that right?

Reply:
Well, I wanted to filter out `DISABLED` status to call `task.complete()` only when it's really needed. WDYT?